### ISSUE TITLE
high rate of stales from OpenCL  2.1 AMD-APP (3224.5)

### ISSUE THREAD

**Issue by jGDarkness** [PhoenixMinerDevTeam/PhoenixMiner#32]
Submitted about 300 shares with 47 stales over about 2 hours or so.

Console window shows the following message on startup:

> PhoenixMiner 5.5c Windows/msvc  -  Release Build
> --------------------------------------------------------
> 
> Unknown OpenCL driverr version! Hashrate and stale shares may suffer
> OpenCL platform: OpenCL 2.1 AMD-APP (3224.5)
> Available GPUs for mining:
> GPU1: AMD Radeon RX 5700 (pcie 12), OpenCL 2.0, 8 GB VRAM, 36 CUs
> Eth: the pool list contains 2 pools (2 from command-line)
> Eth: primary pool: _i've ommitted my pool from this issue report_
> Starting GPU mining
> GPU1: AMD driver 21.3.1
> ...
> ...
> GPU1: Using generic OpenCL kernels (device name 'gfx1010')

**Comment by DeeDeeRanged** [comment]
Do you have a mixed rig AMD and Nvidia?

**Comment by jGDarkness** [comment]
No. At the time I reported the issue, I was running a single AsRock AMD Radeon RX 5700 (Reference Design) Card. I was also using the '-amd' argument. I had no other cards connected to the machine.

**Comment by DeeDeeRanged** [comment]
Had a Nvidia card previously? as it strange that line "GPU1: Using generic OpenCL kernels (device name 'gfx1010')" as that would suggest there is something of Nvidia still floating around.

Anyway do you still have the issue or has it disappeared?

**Comment by jGDarkness** [comment]
I suppose that is possible I had just upgraded to Gigabyte X570 Aorus Master and a Ryzen 9 3950X with fresh windows install. I have an EVGA GTX 1650 Super, but it was never recognized by Windows, and NVIDIA drivers would not install as it could not detect the 1650. I was indexing the GPU to run counting from 1 instead of 0 (just for the heck of it). 

I have temporarily resolved the issue with a Radeon driver roll back to 21.1.1. I suspect it is possibly as simple as the 21.3.x not being officially supported yet, but I don't have any real way of knowing that. 

Currently, I am running PhoenixMiner 5.5c on Radeon 21.1.1 drivers with the RX 5700, and I have two RX 550s running in NBMiner on etchash, all of which have been running virtually non stop since I submitted this issue.

**Comment by mvandergrift** [comment]
Had the exact issue and used the same workaround as @jGDarkness. 

Definitely appears to be that the latest AMD driver is unsupported. Verified this across a variety of 5700xt based cards both on both single & multiple gpu rigs. No nvidia hardware was present yet the generic gfx1010 kernel was selected.  Stale shares increased across several pools from aprox. 0-1% to 8-12% of total.

**Comment by DeeDeeRanged** [comment]
On one system I have a gtx 1070 and get hardly a stale hash, but my other one with a rx 580 got frequently stale shares with phoenixminer but also with teamredminer so I am using now gminer 2.51 with my rx 580 although the hashrate is a bit lower it seems to be more stable maybe 1 or 2 stales within 24 hrs.

**Comment by Hotdog-Vendor** [comment]
My 6800xt uses gfx1030 gets basically 0 stales now. Running driver 21.1.1 and phoenix 5.5c. Have done DDU many times its just what it uses. I had newer drivers on but got 20% stale in phoenix (i think it reported wrong) 5% at pool. Now i get 0. Also changing to older drivers my hash is 64mh/s stable not bouncing from 63.9 to 65.4 etc.
